### ISSUE TITLE
Balance column of stock history does not always show correct running total

### ISSUE THREAD

**Issue by jmiranda** [openboxes/openboxes#122]
TL;DR This is a known issue with the Balance column of the Stock History report (only).

Unlike the quantity on hand calculation, the Balance column calculation is done inline (in the GSP - yuck) and contains somewhat hacky logic that does not take all edge cases into account. For example, the attached screenshot shows a Balance of 6500 on Aug 5. This is because the calculation does not zero out lot number 130221 that was recorded on May 18. The QoH calculation does handle this, but again the Stock History was hastily put together and needs some work.
![amoxicillin 125mg 5ml powder for suspension 100ml bottle - stock history](https://cloud.githubusercontent.com/assets/1162935/11187602/af0a6e2c-8c4c-11e5-87e6-bf3ad8941b95.png)
![amoxicillin 125mg 5ml powder for suspension 100ml bottle](https://cloud.githubusercontent.com/assets/1162935/11187603/af140086-8c4c-11e5-9b44-ddfe8b684473.png)

**Comment by jmiranda** [comment]
Finally found some time to look into this issue. Fixed it by setting the balance to 0 whenever we encounter a product inventory transaction.
![ferrous sulphate 200 mg tablet - fixed](https://cloud.githubusercontent.com/assets/1162935/16305085/ecdc9a3a-391d-11e6-8cab-15e04ee88de4.png)
![ferrous sulphate 200 mg tablet](https://cloud.githubusercontent.com/assets/1162935/16305086/ecdeb144-391d-11e6-9624-6ef22881e666.png)
